src/include/daos/lru.h

```diff
@@ -32,66 +47,47 @@
 struct daos_llink;
 struct daos_llink_ops {
 	/** Mandatory: lru reference free callback */
-	void	(*lop_free_ref)(struct daos_llink *llink);
+	void	 (*lop_free_ref)(struct daos_llink *llink);
 	/** Mandatory: If ref not found, allocate and return ref */
-	int	(*lop_alloc_ref)(void *key, unsigned int ksize,
-				 void *args, struct daos_llink **link);
+	int	 (*lop_alloc_ref)(void *key, unsigned int ksize,
+				  void *args, struct daos_llink **link);
 	/** Mandatory: Compare keys callback for LRU */
-	bool	(*lop_cmp_keys)(const void *key, unsigned int ksize,
-				struct daos_llink *link);
+	bool	 (*lop_cmp_keys)(const void *key, unsigned int ksize,
+				 struct daos_llink *link);
+	/** Mandatory: Get key's hash callback for LRU */
+	uint32_t (*lop_rec_hash)(struct daos_llink *link);
 	/** Optional print_key function for debugging */
-	void	(*lop_print_key)(void *key, unsigned int ksize);
+	void	 (*lop_print_key)(void *key, unsigned int ksize);
 };
 
 struct daos_llink {
-	/* LRU hash link */
-	d_list_t		ll_hlink;
-	/* LRU queue link */
-	d_list_t		ll_qlink;
-	/* Ref count for this reference */
-	unsigned int		ll_ref:30;
-	/** has been evicted */
-	unsigned int		ll_evicted:1;
-	/**
-	 * ops to allocate and free reference
-	 * for this llink.
-	 */
-	struct daos_llink_ops	*ll_ops;
+	d_list_t		 ll_link;	/**< LRU hash link */
+	d_list_t		 ll_qlink;	/**< Temp link for traverse */
+	unsigned int		 ll_ref:30,	/**< refcount for this ref */
+				 ll_evicted:1;	/**< has been evicted */
+	struct daos_llink_ops	*ll_ops;	/**< ops to maintain refs */
 };
 
 /**
- * LRU cache implementation using d_hash_table
- * and d_list_t
+ * LRU cache implementation using d_hash_table and d_list_t
  */
 struct daos_lru_cache {
-	/* Provided cache size */
-	uint32_t		dlc_csize;
-	/* # idle items in the LRU */
-	uint32_t		dlc_idle_nr;
-	/* # busy items in the LRU (referenced by caller) */
-	uint32_t		dlc_busy_nr;
-	/* Queue head, holds idle refs (no refcnt) */
-	d_list_t		dlc_idle_list;
-	/** list head of busy items in the LRU */
-	d_list_t		dlc_busy_list;
-	/* Holds all refs but needs lookup */
-	struct d_hash_table	dlc_htable;
-	/* ops to allocate and free reference */
-	struct daos_llink_ops	*dlc_ops;
+	uint32_t		 dlc_csize;	/**< Provided cache size */
+	uint32_t		 dlc_count;	/**< count of refs in cache */
+	struct d_hash_table	 dlc_htable;	/**< Hash table for all refs */
+	struct daos_llink_ops	*dlc_ops;	/**< ops to maintain refs */
 };
 
 /**
  * Create a DAOS LRU cache
  * This function creates an LRU cache in DRAM
  *
- * \param bits		[IN]	power2(bits) is the size
- *				of the LRU cache
+ * \param bits		[IN]	power2(bits) is the size of the LRU cache
```

Comment:
Not a regression but this isn't doxygen format in this file.  Now might be a good time to bring it inline with all the other changes here.

Reply:
Agree. Will do...